src/batou_ext/file.py

```diff
@@ -109,40 +139,58 @@ def update(self):
             candidates = self._list_removals()
             if not candidates:
                 batou.output.annotate("Nothing to remove.")
-            elif self.use_systemd_run_async_cleanup:
-                # Spawns an systemd-run cleanup job with custom args.
-                candidates = [os.path.join(self.dir, c) for c in candidates]
-                rm_cmd = [
-                    "rm",
-                    "-rfv",
-                    *candidates,  # consider: ARG_MAX is limited
-                ]
-                extra_args = self.systemd_extra_args or []
-                cmd = [
-                    "nohup",
-                    "systemd-run",
-                    "--unit",
-                    f"batou-cleanup-{self.prefix}",
-                    "--user",
-                    *extra_args,
-                    *rm_cmd,
-                    "&",
-                ]
+
+            for c in candidates:
                 batou.output.annotate(f"Removing: {candidates}")
-                batou.output.annotate(f"    {cmd}")
-                try:
-                    subprocess.run(cmd, check=True)
-                except subprocess.CalledProcessError as e:
-                    batou.output.error(f"Failed to remove: {e}")
-            else:
-                for el in candidates:
-                    batou.output.annotate("Removing: {}".format(el))
-                    try:
-                        if os.path.isdir(el):
-                            shutil.rmtree(el)
-                        else:
-                            os.remove(el)
-                    except OSError as e:
-                        batou.output.error(
-                            f'Failed to remove "{el}": {e.strerror}'
-                        )
+                self.trash.discard(c)
+
+
+class DeploymentTrash(batou.component.Component):
+    """
+    A trash folder that is regularly cleaned up by systemd's tmpfiles.
+    Files and Folders can be moved in here for asynchronous deletion that is not tied to the deployment.
+    Due to this component being commonly used to offload deleting very large files, it also supports adding IOPS limits.
+
+
+    ```python
+    self.trash = DeploymentTrash(read_iops_limit = 250, write_iops_limit = 250)
+    self += self.trash
+
+    # files that are created here should not be part of the deployment but rather a sideproduct
+    # for example a cache directory or build artefacts that are generated by some script
+    large_folder = do_something_that_generates_a_lot_of_byproducts()
+    self.trash.discard(large_folder)
+    ```
+    """
+
+    read_iops_limit = 100
```

Comment:
@zagy @PhilTaken is it possible to have these limits configuratable? We currently use 50 in our manual process because 100 caused downtimes again. So this value would be needed to fine tune the process.

Reply:
The Limits are configurable, for example via `self.trash = DeploymentTrash(read_iops_limit = 250, write_iops_limit = 250)` when instancing the Component.